### ISSUE TITLE
Godot window crashes and becomes unresponsive [Linux Gnome]

### ISSUE THREAD

**Issue by vvvvvvitor** [godotengine/godot#74850]
### Godot version

4.0flatpak + all 4.0 betas

### System information

Fedora (Linux), Wayland, Intel® Core™ i3-1005G1 × 4

### Issue description

Godot crashes and becomes unresponsive, only being able to be closed via system monitor.
[Gravação de tela de 2023-03-12 23-12-28.webm](https://user-images.githubusercontent.com/119557151/224592994-459d8491-855f-41de-ae01-3ea7b3747549.webm)

### Steps to reproduce

Happens randomly.

### Minimal reproduction project

Happens with any project.

**Comment by Calinou** [comment]
Can you reproduce this after enabling **Single Window Mode** in the editor settings then restarting the editor?

**Comment by A404M** [comment]
Ooook IDK why my issue got closed because of duplicated and this issue is 3 days after my issue :)

**Comment by A404M** [comment]
Ooook IDK why my issue got closed because of duplicated and this issue is 3 days after mine :)
But anyway, the solution you suggested isn't working for me.
And I have to say that my problem isn't actually the same with this problem, my problem is when I resize the godot window my whole desktop stops working even the mouse (not like what is showed in this video) and after 4 or 5 seconds my system starts working again.

**Comment by ywmaa** [comment]
I have the same problem as @A404M 
not sure if it is exactly the same as the OP

**Comment by vnen** [comment]
Might be the same as #74548

**Comment by A404M** [comment]
> Might be the same as #74548

This might be. but my problem is something else. Can someone reopen my issue and know the difference between mine and this issue?

**Comment by vvvvvvitor** [comment]
> Can you reproduce this after enabling **Single Window Mode** in the editor settings then restarting the editor?

Currently testing and so far it's been working fine, it's really random though so I wouldn't use that as proof that it stops when in Single Window Mode.

**Comment by Kapouett** [comment]
This happened to me (only once) while resizing the game's window while testing a project. Godot filled my entire RAM as well as some of the swap before I could kill it.

**Comment by capnm** [comment]
This just happened during restarts after the _single window mode_ option. (Vulkan API 1.3.238 - Forward+ Ubuntu Gnome, v4.0.stable) :ocean: 

<details><summary>Screenshot</summary>

![grafik](https://user-images.githubusercontent.com/4047289/225478602-d606225b-897a-41fa-9fea-a8e2fd6be17f.png)
Started with godot4 -e in the project directory, finished with ctrl+c, no errors, nothing ...
</details>

**Comment by A404M** [comment]
https://github.com/godotengine/godot/issues/74749#issuecomment-1483994889
My issue got solved thanks to new update

**Comment by wrn4** [comment]
I get random crashes on linux with godot 4.0.1. Using arch linux and official bin from community repo. XFCE as desktop environment.

<details><summary>Error message</summary>
<p>

[xcb] Unknown request in queue while dequeuing
[xcb] You called XInitThreads, this is not your fault
[xcb] Aborting, sorry about that.
godot: xcb_io.c:175: dequeue_pending_request: Assertion `!xcb_xlib_unknown_req_in_deq' failed.

</p>
</details>

**Comment by akien-mga** [comment]
That last error sounds like a duplicate of #69352, which should be fixed by libx11 1.8.4. Can you check which version you have installed?

**Comment by wrn4** [comment]
multilib/lib32-libx11 1.8.4-1 (632.7 KiB 1.3 MiB) (Installed)
    X11 client-side library (32-bit)
extra/libx11 1.8.4-1 (2.0 MiB 10.0 MiB) (Installed)
    X11 client-side library


I have previously ignored this package( 1.8.2 or 1.8.3 i believe) since it caused crashes in some games...but that was only the 32-bit version. I can try to downgrade it.

**Comment by WhalesState** [comment]
it happened to me many times on Ubuntu 20.04.6 LTS

**Comment by darksylinc** [comment]
This sounds like it should've been fixed by PR #80571

That bug can and will cause freezes, crashes and unresponsiveness. Thus it may be worth to try latest dev build to see if the problem went away.

**Comment by vvvvvvitor** [comment]
It just corrupted one of my projects. :)

**Comment by Zireael07** [comment]
@vvvvvvitor This issue is about crashes/unresponsiveness, not file corruption.

**Comment by capnm** [comment]
> This sounds like it should've been fixed by PR #80571
> 
> That bug can and will cause freezes, crashes and unresponsiveness. Thus it may be worth to try latest dev build to see if the problem went away.

I have still (random) occasional issues on current master tip. In Godot-editor after making some changes and hitting the `>` button.  I get the `[Saving Scene 100%]`  window \+ the editor window in deadlock state...

**Comment by vvvvvvitor** [comment]
> @vvvvvvitor This issue is about crashes/unresponsiveness, not file corruption.

It happened to me again and this time it corrupted my project, so it might be related to it? Unless it's a completely different bug.

**Comment by vvvvvvitor** [comment]
> > This sounds like it should've been fixed by PR #80571
> > That bug can and will cause freezes, crashes and unresponsiveness. Thus it may be worth to try latest dev build to see if the problem went away.
> 
> I have still (random) occasional issues on current master tip. In Godot-editor after making some changes and hitting the `>` button. I get the `[Saving Scene 100%]` window + the editor window in deadlock state...

I believe this might be this other issue #79682

**Comment by capnm** [comment]
> I believe this might be this other issue #79682

Honestly, I don't know. I newer used the KDE desktop or a Godot flatpak. It "freezes" for me in gnome x11, sometimes when launching the app from the editor UI or when opening the editor after deleting the Gogot caches. Mostly when running some heavy background jobs...

**Comment by MatiasVME** [comment]
Godot 4.3 crash wayland (Dev Snapshot 4)

IDK if the same bug :S 

https://clips.twitch.tv/CuteGentleTildeChocolateRain-4rjy-l0o4qe6l9du


```
Resizing the window from (1705, 939) to (1705, 939) (fractional scale x1.000000).
libdecor frame on configure rect [P: (0, 0), S: (1705, 939)]
libdecor frame on configure rect [P: (0, 0), S: (1705, 939)]
Resizing the window from (1702, 939) to (1702, 939) (fractional scale x1.000000).
libdecor frame on configure rect [P: (0, 0), S: (1702, 939)]
libdecor frame on configure rect [P: (0, 0), S: (1702, 939)]
Resizing the window from (1701, 939) to (1701, 939) (fractional scale x1.000000).
libdecor frame on configure rect [P: (0, 0), S: (1701, 939)]
Resizing the window from (1701, 928) to (1701, 928) (fractional scale x1.000000).
libdecor frame on configure rect [P: (0, 0), S: (1701, 928)]
Resizing the window from (1699, 928) to (1699, 928) (fractional scale x1.000000).
libdecor frame on configure rect [P: (0, 0), S: (1699, 928)]
Resizing the window from (1699, 927) to (1699, 927) (fractional scale x1.000000).
libdecor frame on configure rect [P: (0, 0), S: (1699, 927)]
Resizing the window from (1698, 927) to (1698, 927) (fractional scale x1.000000).
libdecor frame on configure rect [P: (0, 0), S: (1698, 927)]
Resizing the window from (1697, 926) to (1697, 926) (fractional scale x1.000000).
libdecor frame on configure rect [P: (0, 0), S: (1697, 926)]
ERROR: Wayland client error code 32.
   at: _poll_events_thread (platform/linuxbsd/wayland/wayland_thread.cpp:2542)
ERROR: Caller thread can't call this function in this node (/root). Use call_deferred() or call_thread_group() instead.
   at: propagate_notification (scene/main/node.cpp:2353)

================================================================
handle_crash: Program crashed with signal 4
Engine version: Godot Engine v4.3.dev4.official (df78c0636d79c9545a283e0e2a926d623998cc27)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
ERROR: FATAL: Index p_index = 1 is out of bounds (size() = 0).
   at: get (./core/templates/cowdata.h:205)
matiasvme@matias-gf65thin10sdr:~/Desktop$
```

**Comment by Rindbee** [comment]
It seems that this may be caused by using member variables that are not initialized correctly. I'm not so sure.
